ProfilerGUI/Source/Runner/ProfilerRunner.cs

```diff
@@ -31,21 +31,17 @@ public ProfilerRunner(ProfilerConfiguration configuration)
         /// </summary>
         public void RunAsynchronously()
         {
-            string profilerDllPath = LocateProfilerDll();
-            if (!File.Exists(profilerDllPath))
-            {
-                logger.Error("The profiler DLL was not found at {profilerPath}. Cannot profile the application", profilerDllPath);
-                return;
-            }
+            List<(string, string)> environmentVariables;
 
-            List<(string, string)> environmentVariables = new List<(string, string)>()
-                {
-                    (ProfilerConstants.ProfilerIdEnvironmentVariable, ProfilerConstants.ProfilerGuid),
-                    (ProfilerConstants.ProfilerPathEnvironmentVariable, profilerDllPath),
-                    (ProfilerConstants.EnableProfilingEnvironmentVariable, "1"),
-                    (ProfilerConstants.TargetDirectoryEnvironmentVariable, configuration.TraceTargetFolder),
-                    (ProfilerConstants.LightModeEnvironmentVariable, "1"),
-                };
+            try
+            {
+                environmentVariables = DetermineEnvironmentVariables(configuration);
+            }
+            catch (Exception e)
+            {
+                logger.Error(e.Message);

```

Comment:
maybe add some message where the error occurred (loading env vars). On the other hand I am not sure if we could sift down in below try..catch?

Reply:
I merged the two tries, looks much cleaner and makes perfect sense.